### PR TITLE
Feature/search client changes

### PR DESCRIPTION
**Pull request by merk**
This PR brings the SearchClientStub into line with Search 3.0 changes.

Hard BC break for anything using the SearchClient, but thats why we're in 0.0.x version territory

### CODE REVIEW

File: src/Stubs/BaseStub.php [hunk 1]
```diff
@@ -82,6 +82,16 @@ public function getCalls(string $method): array
         return $this->calls[$method] ?? [];
     }
 
+    /**
+     * Resets the stub's record of all calls.
+     *
+     * @return void
+     */
+    public function resetStub(): void
```
Comment (edwardmurrell-loyaltycorp) [inline src/Stubs/BaseStub.php]: shouldn't stubs be single use?

Comment (merk) [reply src/Stubs/BaseStub.php]: Its a good question - we're using the SearchClientStub inside functional tests where we need to be able to reset them into an empty state after the arrange phase of a test (setting up precondition entities, we only want to assert against changes made during the act phase)

Comment (merk) [reply src/Stubs/BaseStub.php]: I moved the function from SearchClientStub to Stub thinking it'd be helpful in a general case but I'm happy to move it back

Comment (edwardmurrell-loyaltycorp) [reply src/Stubs/BaseStub.php]: I think it's ok, because it's not the generic stub used for everything.

Does it need to clear `$responses` also?

Comment (merk) [reply src/Stubs/BaseStub.php]: Potentially, but that opens the can of worms about should the stub be reconfigurable and set new responses or not?

Maybe rename the method to resetStubCalls() to make it clear it is only resetting the recorded calls?

Comment (edwardmurrell-loyaltycorp) [reply src/Stubs/BaseStub.php]: True. I think that can of worms answers if it should go back into the SearchClientStub then.

Comment (merk) [reply src/Stubs/BaseStub.php]: $calls is private, decided to rename to resetStubCalls() to make it clear only the calls are reset.

**Pull request merged**